docs/specs/sdk.md

```diff
@@ -0,0 +1,180 @@
+# SDK
+
+This document specifies the user experience of the SDK.
+
+## Container, container name and storage
+
+An instance of Container is the entrypoint of the SDK.
+Unless otherwise specified, `authgear` is an instance of Container.
+`authgear.foobar()` means calling the `foobar()` method of the container.
+
+Every container has a name.
+If no name is given, the name is `"default"`.
+
+Everything stored by a container is scoped by its name.
+Therefore, if the developer wants to implement multiple account,
+they must create different container instances with distinct names.
+
+## The use of persistent storage
+
+The SDK by default will store refresh tokens in persistent storage.
+In some applications such as banking applications,
+It is more preferable to have ephemeral refresh token that are only stored in memory.
+
+The SDK can be configured to store refresh token in memory only.
+
+## Logout
+
+Logout is provided by the following API
+
+```typescript
+/**
+ * This function can be called when the user has logged in.
+ *
+ * Depending on whether biometric authentication is enabled or not,
+ * this function behaves differently.
+ *
+ * When biometric authentication is disabled,
+ * calling this function will allow logging in as a different user.
+ *
+ * When biometric authentication is enabled,
+ * calling this function will ONLY terminate the current session.
+ * The last logged in user will NOT be forgotten.
+ * Logging in a different user will be disallowed in the web UI.
+ * If switching different user is desired,
+ * disableBiometric() must be called.
+ *
+ */
+function logout(): Promise<void>;
+```
+
+## Biometric authentication
+
+Biometric authentication is provided by the following API
+
+```typescript
+/**
+ * This function can be called at anytime.
+ * This function reports whether biometric authentication is available.
+ */
+function isBiometricAvailable(): Promise<boolean>;
```

Comment:
What does this meant? Is "available" meant the biometric setting is enabled or not in the server side? (If yes, maybe better specified here)

Reply:
This is a feature checking on the device. For iOS devices, this function will probably return true. But for Android devices, this function reports whether the device support biometric. Let me clarify.

Reply:
Our SDK does NOT support checking whether a particular feature is enabled on the server.

First, the Authgear server itself does not have an endpoint to get the configuration. Only the portal server can do this. The SDK communicates with the Authgear server only, not the portal server.

Second, the developer should have known which feature they have enabled, and only use the subset of API of SDK to complete their job.

Reply:
Ic ok. I was wondering too, and feel strange why we have such a api call.